src/stan_files/functions/hazard_functions.stan

```diff
@@ -0,0 +1,155 @@
+  /**
+  * Log hazard for exponential distribution
+  *
+  * @param eta Vector, linear predictor
+  * @return A vector
+  */
+  vector exponential_log_haz(vector eta) {
+    return eta;
+  }
+
+  /**
+  * Log hazard for exponential distribution; AFT parameterisation
+  *
+  * @param af Vector, acceleration factor at time t
+  * @return A vector
+  */
+  vector exponentialAFT_log_haz(vector af) {
+    return log(af);
+  }
+
+  /**
+  * Log hazard for Weibull distribution
+  *
+  * @param eta Vector, linear predictor
+  * @param t Vector, event or censoring times
+  * @param shape Real, Weibull shape
+  * @return A vector
+  */
+  vector weibull_log_haz(vector eta, vector t, real shape) {
+    vector[rows(eta)] res;
+    res = log(shape) + (shape - 1) * log(t) + eta;
```

Comment:
This should just be `return log(shape) + (shape - 1) *  log(t) + eta;`. Assigning to `res` just slows things down and possibly overpromotes. Similar things below.

Reply:
Ok cheers, have now removed all the redundant assignments in this file.